addons/point_of_sale/models/account_move.py

```diff
@@ -17,6 +17,13 @@ def _stock_account_get_last_step_stock_moves(self):
             stock_moves += invoice.sudo().mapped('pos_order_ids.picking_id.move_lines').filtered(lambda x: x.state == 'done' and x.location_id.usage == 'customer')
         return stock_moves
 
+    @api.depends('pos_order_ids')
+    def _compute_amount(self):
+        super(AccountMove, self)._compute_amount()
+        pos_invoices = self.filtered(lambda i: i.pos_order_ids)
```

Comment:
I'd do:

``` python
pos_moves = self.filtered(lambda i: i.type in ['out_invoice', 'out_refund'] and i.pos_order_ids)
for move in pos_moves:
    move.invoice_payment_state = 'paid'
```
Getting the `type` is trivial (and probably already in cache) while `pos_order_ids` is probably not.